client/src/block.tk.usegm.js

```diff
@@ -936,270 +936,240 @@ function domainlegend(tk, block) {
 	// any customizations to the domains (visibility and color), apply to all gm
 	const domainTypes = client.getdomaintypes(block.usegm)
 
-	for (const domaintype of domainTypes) {
-		/*
-		domaintype {}
-		.key: domain.name+domain.description
-		rest of attributes are the same as individual domains
-		*/
+	const links = [
+		{
+			key: 'CDD',
+			makelink: domaintype => `https://www.ncbi.nlm.nih.gov/Structure/cdd/cddsrv.cgi?uid=${domaintype.CDD}`
+		},
+		{
+			key: 'Pfam',
+			makelink: domaintype => `https://www.ebi.ac.uk/interpro/entry/pfam/PF${domaintype.Pfam.substr(4)}`
+		},
+		{
+			key: 'SMART',
+			makelink: domaintype =>
+				`https://smart.embl.de/smart/do_annotation.pl?BLAST=DUMMY&DOMAIN=${domaintype.SMART.substr(5)}`
+		},
+		{
+			key: 'COG',
+			makelink: domaintype => `https://www.ncbi.nlm.nih.gov/Structure/cdd/cddsrv.cgi?uid=${domaintype.COG}`
+		},
+		{
+			key: 'PRK',
+			makelink: domaintype => `https://www.ncbi.nlm.nih.gov/proteinclusters?term=${domaintype.PRK}`
+		},
+		{
+			key: 'pmid', // may support comma-joined ids
+			makelink: domaintype => `https://pubmed.ncbi.nlm.nih.gov/${domaintype.pmid}/`,
+			txt: domaintype => `PubMed ${domaintype.pmid}`
+		},
+		{
+			key: 'url',
+			makelink: domaintype => domaintype.url,
+			txt: () => 'Reference'
+		}
+	]
+
+	const menuOptions = [
+		{
+			label: 'Hide',
+			hide: d => block.usegm.domain_hidden[d.key],
+			callback: d => {
+				for (const m of block.allgm) {
+					m.domain_hidden[d.key] = 1
+				}
+			}
+		},
+		{
+			label: 'Show',
+			hide: d => !block.usegm.domain_hidden[d.key],
+			callback: d => {
+				for (const m of block.allgm) {
+					delete m.domain_hidden[d.key]
+				}
+			}
+		},
+		{
+			label: 'Show All',
+			hide: () => !Object.keys(block.usegm.domain_hidden).length,
+			callback: () => {
+				for (const m of block.allgm) {
+					m.domain_hidden = {}
+				}
+			}
+		},
+		{
+			label: 'Show only',
+			hide: () => Object.keys(block.usegm.domain_hidden).length == domainTypes.length - 1,
+			callback: d => {
+				for (const m of block.allgm) {
+					delete m.domain_hidden[d.key]
+					for (const c of domainTypes) {
+						if (c.key != d.key) m.domain_hidden[c.key] = 1
+					}
+				}
+			}
+		}
+	]
 
-		const row = tk.td_legend.append('div').style('margin', block.legend.vpad + ' 0px 8px 0px')
+	const legendRows = tk.td_legend
+		.selectAll('div')
+		.data(domainTypes)
+		.enter()
+		.append('div')
+		.classed('sjpp-domain-legend-item', true)
+		.style('margin', block.legend.vpad + ' 0px 8px 0px')
+
+	legendRows
```

Comment:
here can be `const clickableRows = legendRows.append('span')` so that no need for `.each()` at line 1029